### PR TITLE
Feature/shared pref manager

### PR DESCRIPTION
**Pull request by selimtoksal**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,6 +226,41 @@ DeepLinkLauncher.show()
...
+  		<td><img src="/art/sharedpref_manager_1.png" width="200" /></td>
+  		<td><img src="/art/sharedpref_manager_2.png" width="200" /></td>
+  		<td><img src="/art/sharedpref_manager_3.png" width="200" /></td>
+  		<td><img src="/art/sharedpref_manager_4.png" width="200" /></td>
+ 	</tr>
+</table>
+
+### Initialize
+
+You need to initialize it first
```
Comment (bilgehankalkan) [inline README.md]:
```suggestion
You need to initialize the library with with your `android.app.Application` instance.
```

File: libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/domain/SharedPrefUpdateTypeValidator.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import com.trendyol.android.devtools.sharedprefmanager.data.SharedPrefValueType
+
+class SharedPrefUpdateTypeValidator {
+
+    fun validate(sharedPrefModel: SharedPrefModel, newValue: String) {
+        when (sharedPrefModel.valueType) {
+            SharedPrefValueType.INT -> {
+                kotlin.runCatching {
+                    newValue.toInt()
+                }.onSuccess {
```
Comment (bilgehankalkan) [inline libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/domain/SharedPrefUpdateTypeValidator.kt]: Empty `onSuccess` blocks can be removed and exceptions can be gathered in a function like below:

```kotlin
fun throwExceptionFor(valueType: SharedPrefValueType) {
    throw IllegalArgumentException("Old value type is: $valueType so new value type must be $valueType.")
}
```

File: libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/ui/ui/theme/Theme.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    onPrimary = Color.White,
+    onSecondary = Color.White,
+    onTertiary = Color.White,
+    onBackground = Color(0xFF1C1B1F),
+    onSurface = Color(0xFF1C1B1F),
+    */
+)
+
+@Composable
+fun DevtoolsTheme(
```
Comment (bilgehankalkan) [inline libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/ui/ui/theme/Theme.kt]:
```suggestion
fun DevToolsTheme(
```

**Review by bilgehankalkan — APPROVED** [review]

File: libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/domain/SharedPrefUpdateTypeValidator.kt (hunk 2 above)
Comment (selimtoksal) [reply libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/domain/SharedPrefUpdateTypeValidator.kt]: done with 8b14e3d43cf31640a2c0a5110f5988b05c794660

File: libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/ui/ui/theme/Theme.kt (hunk 3 above)
Comment (selimtoksal) [reply libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/ui/ui/theme/Theme.kt]: done with 8b14e3d43cf31640a2c0a5110f5988b05c794660

File: README.md (hunk 1 above)
Comment (selimtoksal) [reply README.md]: done with 8b14e3d43cf31640a2c0a5110f5988b05c794660

**Pull request merged**